packages/query-core/src/queryObserver.ts

```diff
@@ -264,6 +264,34 @@ export class QueryObserver<
     return this.#currentResult
   }
 
+  getServerResult(): QueryObserverResult<TData, TError> {
+    const currentResult = this.#currentResult
+
+    if (
+      currentResult.status === 'success' &&
+      this.#currentQuery.state.dataUpdatedAt === 0
+    ) {
+      const pendingResult: QueryObserverResult<TData, TError> = {
+        ...currentResult,
+        status: 'pending',
+        isPending: true,
+        isSuccess: false,
+        isError: false,
+        isLoading: currentResult.fetchStatus === 'fetching',
+        isInitialLoading: currentResult.fetchStatus === 'fetching',
+        data: undefined,
+        error: null,
+        isLoadingError: false,
+        isRefetchError: false,
+        isPlaceholderData: false,
```

Comment:
how did you come up with this list of fields to change? For example, why aren’t fields like `isFetched` or `isStale` being re-set?

I’m wondering how we can make sure to keep this list in-sync if we add new properties to `QueryObserverResult`...